app/services/FlatRateService.scala

```diff
@@ -207,7 +212,7 @@ class FlatRateService @Inject()(val businessService: BusinessService,
 
   def fetchVatStartDate(implicit headerCarrier: HeaderCarrier, currentProfile: CurrentProfile): Future[LocalDate] = {
     vatApplicationService.getVatApplication.map(_.startDate.getOrElse(
-      throw new InternalServerException("[FlatRateService] Unable to fetch VatStartDate")
+      throw MissingAnswerException(missingRegDateSection)
```

Comment:
this gets recovered by the calculated date from eligibility which always exists, not strictly necessary to make it a MissingAnswerException

Reply:
If some logic changes at any point though, this will be covered and work as expected.